modules/task.js

```diff
@@ -0,0 +1,92 @@
+import Element from './element.js';
+
+const elementObject1 = new Element();
+let editId = 0;
+let i = null;
+export default class Task {
+  constructor() {
+    this.taskList = [];
+  }
+
+  addTask = () => {
+    if (editId !== 0 && i !== null) {
+      this.updateTask(elementObject1.titleInput.value, i);
+    } else {
+      const tasks = {
+        title: elementObject1.titleInput.value,
+        completed: false,
+      };
+      this.taskList.push(tasks);
+      let rd = JSON.parse(localStorage.getItem('tasksInfo'));
+      if (rd) {
+        rd.push(tasks);
+      } else {
+        rd = [];
+        rd.push(tasks);
+      }
+      const taskssString = JSON.stringify(rd);
+      localStorage.setItem('tasksInfo', taskssString);
+      this.retrieveData();
+    }
+  }
+
+  updateTask = (text, check) => {
+    const rd = JSON.parse(localStorage.getItem('tasksInfo'));
+    rd.forEach((element) => {
+      if (check === true || check === false) {
+        if (rd[rd.indexOf(element)].title === text.trim()) {
+          rd[rd.indexOf(element)].title = text.trim();
+          rd[rd.indexOf(element)].completed = check;
+        }
+      } else if (check >= 0) {
+        if (rd.indexOf(element) === check) {
+          rd[rd.indexOf(element)].title = text.trim();
+        }
+      }
+    });
+    const taskssString = JSON.stringify(rd);
+    localStorage.setItem('tasksInfo', taskssString);
+    window.location.reload();
+  }
+
+  removeItem = (tasksTitle) => {
+    let taskList = JSON.parse(localStorage.getItem('tasksInfo'));
+    taskList = taskList.filter((element) => element.title !== tasksTitle);
+    const string = JSON.stringify(taskList);
+    localStorage.setItem('tasksInfo', string);
+    window.location.reload();
+  }
```

Comment:
- [ ] I would like to point out the second point as highlighted below 👇 
![image](https://user-images.githubusercontent.com/90222110/228964510-6fc004ea-d1c1-4fcb-8fe9-484ac817f682.png)

This concerns updating each task's index property so that they stay in order and increment properly. Please make sure that after you have implemented the index property, deleting a task will be able to update other task's index 😉